fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -535,19 +558,46 @@ public Resource insert(Resource resource, List<ExtractedParameterValue> paramete
             stmt = connection.prepareCall(stmtString);
             stmt.setString(1, resource.getResourceType());
             stmt.setString(2, resource.getLogicalId());
-            stmt.setBytes(3, resource.getData());
+
+            // Check for large objects, and branch around it.
+            boolean large = FhirSchemaConstants.STORED_PROCEDURE_SIZE_LIMIT < resource.getData().length;
+            if (large) {
+                // Outside of the normal flow we have a BIG JSON or XML
+                stmt.setNull(3, Types.BLOB);
+            } else {
+                // Normal Flow, we set the data
+                stmt.setBytes(3, resource.getData());
+            }
 
             lastUpdated = resource.getLastUpdated();
             stmt.setTimestamp(4, lastUpdated, UTC);
             stmt.setString(5, resource.isDeleted() ? "Y": "N");
             stmt.setInt(6, resource.getVersionId());
             stmt.registerOutParameter(7, Types.BIGINT);
+            stmt.registerOutParameter(8, Types.BIGINT);
 
             stmt.execute();
             long latestTime = System.nanoTime();
             double dbCallDuration = (latestTime-dbCallStartTime)/1e6;
 
             resource.setId(stmt.getLong(7));
+            long versionedResourceRowId = stmt.getLong(8);
+            if (large) {
```

Comment:
&& resource.getData() != null

Reply:
it would have thrown a NPE prior to this.